### ISSUE TITLE
Confusing messages about incompatible packages

### ISSUE THREAD

**Issue by eelsirhc** [conda/conda#8889]
## Current Behavior
The compatibility message from conda is very confusing, and generally wrong, when installing a package that is incompatible with the current environment. Conda used to offer to drastically modify the environment (also annoying at times) but now emits a cryptic message claiming the current environment is incompatible with itself.

### Steps to Reproduce
1. Create a simple Python 3.7 environment
```
conda create -n p37 python=3.7

WARNING conda.base.context:use_only_tar_bz2(632): Conda is constrained to only using the old .tar.bz2 file format because you have conda-build installed, and it is <3.18.3.  Update or remove conda-build to get smaller downloads and faster extractions.
Collecting package metadata (repodata.json): done
Solving environment: done

## Package Plan ##

  environment location: /nobackup/clee25/anaconda/envs/p37

  added / updated specs:
    - python=3.7


The following NEW packages will be INSTALLED:

  _libgcc_mutex      pkgs/main/linux-64::_libgcc_mutex-0.1-main
  ca-certificates    pkgs/main/linux-64::ca-certificates-2019.5.15-0
  certifi            pkgs/main/linux-64::certifi-2019.6.16-py37_0
  libedit            pkgs/main/linux-64::libedit-3.1.20181209-hc058e9b_0
  libffi             pkgs/main/linux-64::libffi-3.2.1-hd88cf55_4
  libgcc-ng          pkgs/main/linux-64::libgcc-ng-9.1.0-hdf63c60_0
  libstdcxx-ng       pkgs/main/linux-64::libstdcxx-ng-9.1.0-hdf63c60_0
  ncurses            pkgs/main/linux-64::ncurses-6.1-he6710b0_1
  openssl            pkgs/main/linux-64::openssl-1.1.1c-h7b6447c_1
  pip                pkgs/main/linux-64::pip-19.1.1-py37_0
  python             pkgs/main/linux-64::python-3.7.3-h0371630_0
  readline           pkgs/main/linux-64::readline-7.0-h7b6447c_5
  setuptools         pkgs/main/linux-64::setuptools-41.0.1-py37_0
  sqlite             pkgs/main/linux-64::sqlite-3.28.0-h7b6447c_0
  tk                 pkgs/main/linux-64::tk-8.6.8-hbc83047_0
  wheel              pkgs/main/linux-64::wheel-0.33.4-py37_0
  xz                 pkgs/main/linux-64::xz-5.2.4-h14c3975_4
  zlib               pkgs/main/linux-64::zlib-1.2.11-h7b6447c_3


Proceed ([y]/n)? 

Preparing transaction: done
Verifying transaction: done
Executing transaction: done
#
# To activate this environment, use
#
#     $ conda activate p37
#
# To deactivate an active environment, use
#
#     $ conda deactivate
```

2. Attempt to install an empty package with a dependency on python 3.6
```
conda install -c chrislee test1

WARNING conda.base.context:use_only_tar_bz2(632): Conda is constrained to only using the old .tar.bz2 file format because you have conda-build installed, and it is <3.18.3.  Update or remove conda-build to get smaller downloads and faster extractions.
Collecting package metadata (repodata.json): done
Solving environment: failed

UnsatisfiableError: The following specifications were found to be incompatible with each other:

  - pip -> python[version='>=3.6,<3.7.0a0']
```

This message, or a variation on it, occurs when installing a package with incompatible requirements to the current environment.  e.g.  This next package is also listed as Python 3.6 only, but installing it makes it look like the environment has become incompatible with itself.

```
conda install -c chrislee dwell
WARNING conda.base.context:use_only_tar_bz2(632): Conda is constrained to only using the old .tar.bz2 file format because you have conda-build installed, and it is <3.18.3.  Update or remove conda-build to get smaller downloads and faster extractions.
Collecting package metadata (repodata.json): done
Solving environment: failed

UnsatisfiableError: The following specifications were found to be incompatible with each other:

  - anaconda==2019.03=py37_0 -> importlib_metadata==0.8=py37_0 -> zipp[version='>=0.3.2']
  - anaconda==2019.03=py37_0 -> mkl-service==1.1.2=py37he904b0f_5
  - importlib_metadata
  - mkl-service
  - pip -> python[version='>=3.6,<3.7.0a0']
  - zipp
```

## Expected Behavior
1. conda should indicate that the incoming package is incompatible, not the existing environment. (i.e. test1 requires python 3.6 but the environment requires 3.7)
2. conda should indicate that the incompatible packages are those about to be installed, not the same packages already installed (i.e. pip to be installed requires python 3.6)

## Environment Information
<details open><summary><code>`conda info`</code></summary><p>

```
     active environment : p37
    active env location : /nobackup/X/anaconda/envs/p37
            shell level : 2
       user config file : /home5/X/.condarc
 populated config files : /home5/X/.condarc
          conda version : 4.7.5
    conda-build version : 3.17.8
         python version : 3.7.3.final.0
       virtual packages : 
       base environment : /home5/X/anaconda37  (writable)
           channel URLs : https://conda.anaconda.org/bioconda/linux-64
                          https://conda.anaconda.org/bioconda/noarch
                          https://conda.anaconda.org/chrislee/linux-64
                          https://conda.anaconda.org/chrislee/noarch
                          https://conda.anaconda.org/auto/linux-64
                          https://conda.anaconda.org/auto/noarch
                          https://repo.anaconda.com/pkgs/main/linux-64
                          https://repo.anaconda.com/pkgs/main/noarch
                          https://repo.anaconda.com/pkgs/r/linux-64
                          https://repo.anaconda.com/pkgs/r/noarch
          package cache : /nobackup/X/anaconda/pkgs
       envs directories : /nobackup/X/anaconda/envs
                          /home5/X/anaconda37/envs
                          /home5/X/.conda/envs
               platform : linux-64
             user-agent : conda/4.7.5 requests/2.21.0 CPython/3.7.3 Linux/4.4.162-94.72.1.20181113 glibc/2.22
                UID:GID : X:X
             netrc file : None
           offline mode : False

```
</p></details>

<details open><summary><code>`conda config --show-sources`</code></summary><p>

```
==> /home5/X/.condarc <==
envs_dirs:
  - /nobackup/X/anaconda/envs
pkgs_dirs:
  - /nobackup/X/anaconda/pkgs
channels:
  - bioconda
  - chrislee
  - auto
  - defaults
```
</p></details>

<details><summary><code>`conda list --show-channel-urls`</code></summary><p>

```
# packages in environment at /nobackup/X/anaconda/envs/p37:
#
# Name                    Version                   Build  Channel
_libgcc_mutex             0.1                        main    defaults
ca-certificates           2019.5.15                     0    defaults
certifi                   2019.6.16                py37_0    defaults
libedit                   3.1.20181209         hc058e9b_0    defaults
libffi                    3.2.1                hd88cf55_4    defaults
libgcc-ng                 9.1.0                hdf63c60_0    defaults
libstdcxx-ng              9.1.0                hdf63c60_0    defaults
ncurses                   6.1                  he6710b0_1    defaults
openssl                   1.1.1c               h7b6447c_1    defaults
pip                       19.1.1                   py37_0    defaults
python                    3.7.3                h0371630_0    defaults
readline                  7.0                  h7b6447c_5    defaults
setuptools                41.0.1                   py37_0    defaults
sqlite                    3.28.0               h7b6447c_0    defaults
tk                        8.6.8                hbc83047_0    defaults
wheel                     0.33.4                   py37_0    defaults
xz                        5.2.4                h14c3975_4    defaults
zlib                      1.2.11               h7b6447c_3    defaults
```
</p></details>

**Comment by eelsirhc** [comment]
# Related issues
1. #3901 - about being verbose during downgrades. not the same issue.
2. #8638 - Similar issue when creating environments and conda issued a warning about incompatible packages but doesn't highlight the incompatibility. 
3. #8574 - Similar. Claimed to be fixed in 4.7.0 and closed as a duplicate of #8638.

**Comment by msarahan** [comment]
Yes, there's a boatload of these issues.  This is a really, really hard problem.  We're trying to improve it, and some improvement was made in 4.7.0, but there's more work to do.
